packages/jsActions/mobile-resources-native/src/file-download/DownloadFile.ts

```diff
@@ -0,0 +1,85 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import { Platform } from "react-native";
+import RNBlobUtil from "react-native-blob-util";
+import FileViewer from "react-native-file-viewer";
+import mimeTypes from "mime";
+
+// BEGIN EXTRA CODE
+function formatPath(...pathArgs: string[]): string {
+    return pathArgs.filter(arg => !!arg).join("/");
+}
+
+function sanitizeFileName(name: string) {
+    /* eslint-disable-next-line no-control-regex */
+    return name.replace(/[<>"?:|*/\\\u0000-\u001F\u007F]/g, "_");
+}
+
+async function getUniqueFilePath(path: string) {
+    if (!(await RNBlobUtil.fs.exists(path))) {
+        return path;
+    }
+
+    const extentionIndex = path.lastIndexOf(".");
+    const extension = path.substring(extentionIndex);
+    let insertionIndex = path.lastIndexOf("(");
+    if (insertionIndex === -1) {
+        insertionIndex = extentionIndex;
+    }
```

Comment:
Two things:
- We should not check for `(`. If the original file name already contains anything with parentheses, this could put the added number in a random place and delete part of the filename. We should always insert the number just before the extension dot (if present), even if the filename already contains one. If we're dealing with just the filename (see the next point), this is simply `const insertionIndex = path.includes(".") ? path.lastIndexOf(".") : path.length;`. (This allows you to get rid of the misspellied "extention" as well. :))
- This does not deal with the problem of path separators yet. This string manipulation is unsafe if the path contains a `.` somewhere in a folder name. This can easily be fixed by doing the `formatPath` here and passing the directory and the filename separately, as I suggested.

Reply:
so, if the file name is `falula(1).png` what should it become? this way it will be `falula(1)(1).png` and the one after that will be `falula(1)(1)(1).png`, is this what you mean?

Reply:
✨